### PR TITLE
Add self-register for integration tests

### LINKED ISSUE #883: Test runner: Collect integration tests dynamically
We statically register tests in the runner though this simple string match. 

https://github.com/infinyon/fluvio/blob/8a04312265228686ef9bd752f25c1472a45bd6b1/tests/runner/src/main.rs#L38-L49

However, this isn't a strategy that will scale to our needs. What we need is a way to collect and link the tests, which are marked with the `#[fluvio_test]` proc macro in the test runner's `tests` module.

Cargo test's `#[test]` macro actually creates a new `main` for annotated tests to run within. Not exactly what we need because we need granular control of setup and teardown. What we want is a way to decentralize the registration of tests using the `#[fluvio_test]` codegen

The [inventory](https://crates.io/crates/inventory) crate will help with building a compile-time list of function pointers that we can use in the test runner.

### PR DESCRIPTION
**Pull request by tjtelan**
Use `inventory` crate and modified proc macro to create a centralized register of pointers to subcommand parsers and test functions.

Resolves #883

### CODE REVIEW

File: tests/runner/Cargo.toml [hunk 1]
```diff
@@ -25,11 +25,13 @@ fluvio-system-util = { version = "0.1.0", path = "../../src/utils" }
 # Fluvio test framework Attribute macro
 fluvio-integration-derive = { path = "./src/fluvio-integration-derive" }
 fluvio-test-util = { path = "./src/utils" }
+#fluvio-integration-tests = { path = "./src/tests"}
 
 syn = { version = "1.0", features = ["full"]}
 
 rand = "^0.8"
```
Comment (sehz) [inline tests/runner/Cargo.toml:32]: do we need `^` prefix

File: tests/runner/Cargo.toml [hunk 2]
```diff
@@ -25,11 +25,13 @@ fluvio-system-util = { version = "0.1.0", path = "../../src/utils" }
...
 md-5 = "^0.9"
 serde = "^1.0"
 serde_json = "^1.0"
+inventory = "0.1"
 tokio = { version = "0.2.21", features = ["macros"] }
```
Comment (sehz) [inline tests/runner/Cargo.toml:37]: update to use latest version

File: tests/runner/src/main.rs [hunk 3]
```diff
@@ -56,37 +75,19 @@ fn main() {
...
-            //}
-            _ => panic!("Test not found"),
-        };
+        let test = inventory::iter::<FluvioTest>
+            .into_iter()
+            .find(|t| t.name == test_name.as_str())
+            .expect("Test not found");
+
+        // Run the test
+        (test.test_fn)(fluvio_client.clone(), test_case);
```
Comment (sehz) [inline tests/runner/src/main.rs:84]: need error handling?

File: tests/runner/src/tests/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+[package]
+name = "fluvio-integration-tests"
+version = "1.0.1"
+authors = ["Fluvio Contributors <team@fluvio.io>"]
+edition = "2018"
+autotests = false
+
+[dependencies]
+log = "0.4.8"
+bytes = "0.5.3"
```
Comment (sehz) [inline tests/runner/src/tests/Cargo.toml:10]: use latest version of dependencies

File: tests/runner/src/tests/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+futures-lite = "1.11.0"
+structopt = "0.3.5"
+async-trait = "0.1.21"
+
+## Fluvio dependencies
+fluvio = { path = "../../../../src/client" }
+#fluvio-types = { version = "0.2.0", path = "../../src/types" }
```
Comment (sehz) [inline tests/runner/src/tests/Cargo.toml:17]: remove comments

File: tests/runner/src/tests/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+#
+## Fluvio test framework Attribute macro
+fluvio-integration-derive = { path = "../fluvio-integration-derive" }
+fluvio-test-util = { path = "../utils" }
+#fluvio-tes`t-opts-builder = { path = "./src/fluvio-test-opts-builder" }
+#fluvio-integration-tests = { path = "tests"}
+
+syn = { version = "1.0", features = ["full"]}
+
+rand = "^0.8"
```
Comment (sehz) [inline tests/runner/src/tests/Cargo.toml:33]: remove `^`  prefix unless it is needed

File: tests/runner/src/utils/Cargo.toml [hunk 7]
```diff
@@ -11,12 +11,12 @@ futures-lite = "1.11.0"
...
-
 syn = "1.0"
 serde = "1.0"
 serde_json = "1.0"
 quote = "1.0"
 proc-macro2 = "1.0"
+inventory = "0.1"
 
 fluvio = { path = "../../../../src/client" }
 fluvio-future = { version = "0.1.0", features = ["task", "timer", "subscriber", "fixture"] }
```
Comment (sehz) [inline tests/runner/src/utils/Cargo.toml:22]: futures should be `0.2.0`

**Review by sehz — CHANGES_REQUESTED** [review]
LGTM.  Minor nits

File: tests/runner/src/main.rs (hunk 3 above)
Comment (tjtelan) [reply tests/runner/src/main.rs:59]: We're catching the panic in a very simple manner at the moment at L44-L47. #880 will introduce more error handling so we can implement reporting

**Comment by sehz** [comment]
nice: 
```
flvt smoke2
error: 'smoke2' isn't a valid value for '<test-name>'
        [possible values: concurrent, producer_stress, smoke]

        Did you mean 'smoke'?

USAGE:
    flv-test <test-name> --replication <replication> --spu <spu> --timeout <timeout> --tls-user <tls-user> --topic-name <topic-name>

```

**Comment by sehz** [comment]
Rebase this then can merge

**Pull request merged**